ui/jstests/test_manage_taxonomies.jsx

```diff
@@ -32,17 +45,79 @@ define(['QUnit', 'jquery', 'setup_manage_taxonomies', 'reactaddons',
       }
     ]
   };
+  var vocabularies = [
+    {
+      "vocabulary": vocabulary,
+      "terms": vocabulary.terms
+    }
+  ];
+  var saveVocabularyResponse;
   function reportError(msg) {
     if (msg) {
       console.error(msg);
     }
   }
+  function updateParent(data) {
+    saveVocabularyResponse = data;
+  }
+  function addTermsComponentRenderTest(assert, rootNode) {
+    // check term title
+    var listOfLinks = React.addons.TestUtils.
+      scryRenderedDOMComponentsWithTag(
+        rootNode,
+        'a'
+    );
+    assert.equal(listOfLinks.length, 2);
+    var linkHeader = listOfLinks[0].getDOMNode();
+    assert.equal(linkHeader.innerHTML, vocabulary.name);
+    //test items
+    var devItems = React.addons.TestUtils.
+      findRenderedDOMComponentWithClass(
+        rootNode,
+        'panel-body'
+    );
+    var itemList = React.addons.TestUtils.
+      scryRenderedDOMComponentsWithTag(
+        devItems,
+        'li'
+    );
+    assert.equal(itemList.length, 2);
+  }
+  function addVocabularyRenderTest(assert, rootNode) {
+    var formNode = React.addons.TestUtils.
+      findRenderedDOMComponentWithClass(
+        rootNode,
+        'form-horizontal'
+      );
+    assert.ok(formNode);
+    //test form submission
+    var inputNodes = React.addons.TestUtils.
+      scryRenderedDOMComponentsWithTag(
+        formNode,
+        'input'
+      );
+    assert.equal(inputNodes.length, 4);
+    inputNodes[0].value = "TestA";
+    inputNodes[1].value = "TestA";
+    React.addons.TestUtils.Simulate.submit(formNode);
+    assert.equal(
```

Comment:
You will need to use `TestUtils.waitForAjax` and `assert.async()` to wait for the POST to complete before finishing the test assertions